src/components/TokenSelection/TokenSelection.tsx

```diff
@@ -123,69 +143,99 @@ const TokenSelection = ({
     );
   }, [sortedInactiveTokens, tokenQuery]);
 
+  useEffect(() => {
+    if (containerRef.current && scrollContainerRef.current) {
+      const { offsetTop, scrollHeight } = scrollContainerRef.current;
+      setOverflow(scrollHeight + offsetTop > containerRef.current.offsetHeight);
+    }
+  }, [
+    containerRef,
+    scrollContainerRef,
+    activeTokens,
+    sortedTokens,
+    allTokens,
+    tokenQuery,
+    width,
+    height,
+  ]);
+
   return (
-    <div>
-      <div className="flex flex-wrap align-middle justify-between">
-        <label className="font-bold text-sm" htmlFor="tokenQuery">
-          Select token
-        </label>
-        <HiX
-          className="light:text-black text-xl cursor-pointer"
-          onClick={closeModal}
-        />
-      </div>
-      <input
+    <Container ref={containerRef} $overflow={overflow}>
+      <TitleContainer>
+        <Title type="h2">{t("common:swap")}</Title>
+        <CloseButton icon="chevron-down" iconSize={1} onClick={onClose} />
+      </TitleContainer>
+      <SearchInput
+        hideLabel
         id="tokenQuery"
         type="text"
+        label="Search name or address"
         value={tokenQuery}
         placeholder="Search name or paste address"
         onChange={(e) => {
-          setTokenQuery(e.target.value);
+          setTokenQuery(e.currentTarget.value);
         }}
-        className="w-full"
       />
-      <div>
-        {filteredTokens.map((token) => (
-          <TokenRow
-            token={token}
-            balance={formatUnits(
-              balances.values[token.address] || 0,
-              token.decimals
-            )}
-            setToken={handleClick}
-            disabled={
-              tokenSelectType === "senderToken"
-                ? token.address === senderToken
-                : token.address === signerToken
-            } // shouldn't be able to select same duplicate token
-            removeActiveToken={removeActiveToken}
-            defaultToken={defaultActiveTokens[chainId!].includes(token.address)}
-            key={`${token.address}`}
-          />
-        ))}
+
+      <ScrollContainer ref={scrollContainerRef}>
+        <Legend>
```

Comment:
IMO the legend should not scroll with the token list.

Reply:
But the legend is for the filteredTokens and not for inactiveTokens. This is something we can discuss with design.